engine/src/main/java/org/terasology/rendering/cameras/PerspectiveCamera.java

```diff
@@ -54,13 +54,11 @@
 
     private Vector3f tempRightVector = new Vector3f();
 
-    public PerspectiveCamera(WorldProvider worldProvider, RenderingConfig renderingConfig) {
+    public PerspectiveCamera(WorldProvider worldProvider, RenderingConfig renderingConfig, DisplayDevice displayDevice) {
         super(worldProvider, renderingConfig);
         this.cameraSettings = renderingConfig.getCameraSettings();
 
-        // TODO: Switch to context.
-        DisplayDevice display = CoreRegistry.get(DisplayDevice.class);
-        display.subscribe(this);
+        displayDevice.subscribe(this);
```

Comment:
Can we subscribe to just the DISPLAY_RESOLUTION_CHANGE property of the displayDevice? Not strictly necessary, I know, but I feel it would help readability and, who knows, one day we might subscribe to the display for other observable properties.